pyproject.toml

```
@@ -0,0 +1,71 @@
+[build-system]
+requires = ["setuptools>=60", "setuptools-scm>=8.0"]
+build-backend = "setuptools.build_meta"
+
+[project]
+name = "emailproxy"
+authors = [
+    {name = "Simon Robinson", email = "simon@robinson.ac"},
+]
+description = """
+Transparently add OAuth 2.0 support to IMAP/POP/SMTP clients that don't \
+support this authentication method.\
+"""
+readme = "README.md"
+license = {file = "LICENSE"}
+classifiers = [
+    "Programming Language :: Python :: 3",
+    "Topic :: Communications :: Email",
+    "Topic :: Communications :: Email :: Post-Office",
+    "Topic :: Communications :: Email :: Post-Office :: IMAP",
+    "Topic :: Communications :: Email :: Post-Office :: POP3",
+    "License :: OSI Approved :: Apache Software License",
+]
+dependencies = [
+    "cryptography",
+
+    # provide the previously standard library module `asyncore`, removed in
+    # Python 3.12 (https://peps.python.org/pep-0594/)
+    "pyasyncore; python_version >= '3.12'",
+
+    # macOS only: output to unified logging
+    "pyoslog>=0.3.0; sys_platform == 'darwin'",
+]
+dynamic = ["version"]
+
+[project.optional-dependencies]
+external-auth = ["prompt_toolkit"]
+gui = [
+    "pillow",  # to create the menu bar icon image from a TTF icon
+    "setuptools",  # for pkg_resources (checking dependency versions)
+    "timeago",  # for displaying the last authenticated activity hint
+
+    # force pystray version with dummy GUI fix for headless deployments
+    # (https://github.com/moses-palmer/pystray/issues/118)
+    "pystray>=0.19.4",
+
+    # force pywebview 4.2.1+ to fix Windows issue with PyInstaller/pythonw
+    # (https://github.com/r0x0r/pywebview/issues/1086)
+    # and a macOS pre-Mojave crash when opening browser windows
+    # (https://github.com/r0x0r/pywebview/pull/1047),
+    # plus a missing macOS dependency issue introduced in 4.1
+    # (https://github.com/r0x0r/pywebview/pull/1154)
+    #
+    # - note that we could do, e.g., platform_release < '18' to allow Linux
+    # platforms more flexibility, but that seems over-the-top
+    "pywebview>=4.2.1",
+
+    # macOS: improve menu bar interaction, provide native notifications and
+    # handle system events
+    "pyobjc-framework-Cocoa; sys_platform == 'darwin'",
+    "pyobjc-framework-SystemConfiguration; sys_platform == 'darwin'",
+]
+
+[project.scripts]
+emailproxy = "emailproxy:App"
+
+[project.urls]
+Homepage = "https://github.com/simonrob/email-oauth2-proxy"
+"Bug Tracker" = "https://github.com/simonrob/email-oauth2-proxy/issues"
+
+[tool.setuptools_scm]
```

I hadn't noticed there was a version specified in the script. This can be used instead, but I'm not sure if a dunder attribute can be referenced (I would expect/hope so)

```suggestion
[tool.setuptools.dynamic]
version = {attr = "emailproxy.__version__"}
```

---

It does appear to work, but the version would need to be `.` separated not `-` separated so it couldn't be ISO 8601

---

This can be adjusted to use `__package_version__` if there were a package version defined by something like:

```python
__package_version__ = '.'.join([str(int(i)) for i in __version__.split('-')])
```